### PR TITLE
feat - Included tests for disk cache

### PR DESCRIPTION
**Pull request by naveensrinivasan**
* **Please check if the PR fulfills these requirements**
- [x] Tests for the changes have been added (for bug fixes/features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Tests for disk cache.

* **What is the current behavior?** (You can also link to an open issue here)
Not many tests for disk cache.

* **What is the new behavior (if this is a feature change)?**
Included tests for disk cache.
Cleaned up tests.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
None

* **Other information**:

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -57,16 +58,35 @@ e2e: build check-env ginkgo
...
+	rm -rf cache
+	mkdir $(OUTPUT)
+	mkdir cache
+	@echo Focusing on these tests $(FOCUS_DISK_TEST)
+	USE_DISK_CACHE=1 DISK_CACHE_PATH="./cache" \
+				   ./scorecard --repo=https://github.com/ossf/scorecard \
+				   --show-details --metadata=openssf  --format json > ./$(OUTPUT)/results.json
+	USE_DISK_CACHE=1 DISK_CACHE_PATH="./cache" ginkgo -p  -v -cover --focus=$(FOCUS_DISK_TEST)  ./e2e/...
+	# Rerun the test with the disk cache
+	USE_DISK_CACHE=1 DISK_CACHE_PATH="./cache" ./scorecard \
```
Comment (inferno-chromium) [inline Makefile:85]: I don't understand the difference from these two commands next from the previous ones. Are previous ones supposed to be run with USE_DISK_CACHE=0 ?

Comment (inferno-chromium) [reply Makefile:85]: Can you also move "./scorecard" to next line, to match other command.

**Review by inferno-chromium — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (naveensrinivasan) [reply Makefile:85]: The whole idea is to start without a cache in the first run and in the second run it should fetch from the cache. 

The goal of this test is to make sure the cache didn't have any issues.

Comment (inferno-chromium) [reply Makefile:85]: Makes sense, thanks for clarifying this.

**Pull request merged**